python/aie/compiler/aiecc/main.py

```diff
@@ -417,18 +429,23 @@ async def run_flow(self):
         self.progress_bar = progress_bar
         progress_bar.task = progress_bar.add_task("[green] MLIR compilation:", total=1, command="1 Worker")
 
+        aiedialect.register_dialect(ctx)
+        mlir_module = Module.parse(self.mlir_module)
```

Comment:
This is slick... Maybe next look at some of the places where this would make life much simpler, for instance where we today use aie-translate to query the device, instead we could just parse the MLIR directly.

Reply:
Yeah that's kind of the point of doing this :)